core/store/src/db.rs

```diff
@@ -751,11 +758,15 @@ fn choose_cache_size(col: DBCol) -> usize {
     }
 }
 
-fn rocksdb_column_options(col: DBCol) -> Options {
+fn rocksdb_column_options(col: DBCol, extra_db_memory: usize) -> Options {
     let mut opts = Options::default();
     opts.set_level_compaction_dynamic_level_bytes(true);
-    let cache_size = choose_cache_size(col);
-    opts.set_block_based_table_factory(&rocksdb_block_based_options(cache_size));
+
+    // In config there is `extra_db_memory_bytes` setting. It specifies extra memory tht can be used
+    // to improve database performance. For now, let's simply add the same cache size for each column.
+    // We can add a better way of using extra memory later.
+    let cache_size = choose_cache_size(col) + extra_db_memory / NUM_DB_COLS;
```

Comment:
no strong opinion here, but it feel that ` extra_db_memory / NUM_DB_COLS`  logic belongs to the caller.